styx-scheduler-service/src/main/java/com/spotify/styx/flyte/FlyteAdminClientRunner.java

```diff
@@ -62,4 +72,46 @@ public FlyteExecution createExecution(final String name, final FlyteExecConf fly
       throw new CreateExecutionException(flyteExecConf, e);
     }
   }
+
+  public void poll(final String project, final String domain, final String name, RunState runState) {
+    Objects.requireNonNull(project);
+    Objects.requireNonNull(domain);
+    Objects.requireNonNull(name);
+    final ExecutionOuterClass.Execution execution =
+        flyteAdminClient.getExecution(project, domain, name);
+    emitFlyteEvents(execution, runState);
+  }
+
+  private void emitFlyteEvents(ExecutionOuterClass.Execution execution, RunState runState) {
+    final Execution.WorkflowExecution.Phase phase = execution.getClosure().getPhase();
+    final FlytePhase flytePhase = FlytePhase.fromProto(phase);
+    try {
+      switch (flytePhase) {
+        case SUCCEEDED:
+          stateManager.receive(Event.terminate(runState.workflowInstance(), Optional.of(0)));
+          break;
+        case FAILED:
+        case ABORTED:
+        case TIMED_OUT:
+          final String flyteCode = execution.getClosure().getError().getCode();
+          final int styxCode = flyteCodeToStyx(flyteCode);
+          stateManager.receive(Event.terminate(runState.workflowInstance(), Optional.of(styxCode)));
+          break;
+      }
+    } catch (IsClosedException e) {
+      return;
+    }
+  }
+
+  private int flyteCodeToStyx(final String flyteCode) {
+      switch (flyteCode) {
```

Comment:
Can you put error in the name so it is more clear that we don't need to account for something to return 0?
```suggestion
  private int flyteErrorCodeToStyx(final String errorCode) {
      switch (errorCode) {
```

Reply:
👍